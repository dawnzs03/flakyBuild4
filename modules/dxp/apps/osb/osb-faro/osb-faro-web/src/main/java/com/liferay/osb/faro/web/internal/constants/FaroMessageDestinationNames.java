/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Marcos Martins
 */
public class FaroMessageDestinationNames {

	public static final String FARO_CHECK_FARO_PROJECTS_MESSAGE_PROCESSOR =
		"liferay/faro_check_faro_projects_message_processor";

	public static final String FARO_CLEAN_DL_FILE_ENTRY_MESSAGE_PROCESSOR =
		"liferay/faro_clean_dl_file_entry_message_processor";

	public static final String FARO_DAILY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_daily_email_report_message_processor";

	public static final String FARO_MONTHLY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_monthly_email_report_message_processor";

	public static final String FARO_WEEKLY_EMAIL_REPORT_MESSAGE_PROCESSOR =
		"liferay/faro_weekly_email_report_message_processor";

}