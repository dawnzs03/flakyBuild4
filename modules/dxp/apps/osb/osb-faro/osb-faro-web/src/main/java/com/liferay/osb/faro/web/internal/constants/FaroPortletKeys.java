/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Shinn Lok
 */
public class FaroPortletKeys {

	public static final String CEREBRO = "cerebro_portlet";

	public static final String FARO = "faro_portlet";

}