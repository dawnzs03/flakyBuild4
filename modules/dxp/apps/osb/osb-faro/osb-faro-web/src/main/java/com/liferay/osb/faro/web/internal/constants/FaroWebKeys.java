/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

/**
 * @author Matthew Kong
 */
public class FaroWebKeys {

	public static final String CEREBRO_ASSETS_URL = "CEREBRO_ASSETS_URL";

	public static final String CEREBRO_TOUCHPOINTS_URL =
		"CEREBRO_TOUCHPOINTS_URL";

}