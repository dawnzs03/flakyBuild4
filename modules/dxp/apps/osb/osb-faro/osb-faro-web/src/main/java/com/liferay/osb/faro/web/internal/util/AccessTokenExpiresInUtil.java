/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

/**
 * @author Ivica Cardic
 */
public class AccessTokenExpiresInUtil {

	public static long getExpiresIn() {
		return _expiresIn;
	}

	public static void removeExpiresIn() {
		_expiresIn = 0;
	}

	public static void setExpiresIn(long expiresIn) {
		_expiresIn = expiresIn;
	}

	private static long _expiresIn;

}