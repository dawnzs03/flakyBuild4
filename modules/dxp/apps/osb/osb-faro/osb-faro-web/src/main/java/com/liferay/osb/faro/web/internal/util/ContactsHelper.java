/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.util;

import com.liferay.osb.faro.engine.client.ContactsEngineClient;
import com.liferay.osb.faro.engine.client.constants.FieldMappingConstants;
import com.liferay.osb.faro.engine.client.model.Individual;
import com.liferay.osb.faro.engine.client.model.IndividualSegment;
import com.liferay.osb.faro.model.FaroProject;
import com.liferay.osb.faro.web.internal.constants.FaroConstants;
import com.liferay.osb.faro.web.internal.model.display.contacts.IndividualDisplay;
import com.liferay.osb.faro.web.internal.model.display.contacts.IndividualSegmentDisplay;
import com.liferay.osb.faro.web.internal.model.display.main.FaroEntityDisplay;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Shinn Lok
 */
@Component(service = ContactsHelper.class)
public class ContactsHelper {

	public FaroEntityDisplay getContactsEntityDisplay(
		FaroProject faroProject, String contactsEntityId, int type) {

		if (type == FaroConstants.TYPE_INDIVIDUAL) {
			Individual individual = _contactsEngineClient.getIndividual(
				faroProject, contactsEntityId, null);

			return new IndividualDisplay(individual);
		}

		IndividualSegment individualSegment =
			_contactsEngineClient.getIndividualSegment(
				faroProject, contactsEntityId, false);

		return new IndividualSegmentDisplay(individualSegment);
	}

	public String getOwnerType(int type) {
		if (type == FaroConstants.TYPE_ACCOUNT) {
			return FieldMappingConstants.OWNER_TYPE_ACCOUNT;
		}
		else if (type == FaroConstants.TYPE_INDIVIDUAL) {
			return FieldMappingConstants.OWNER_TYPE_INDIVIDUAL;
		}

		return FieldMappingConstants.OWNER_TYPE_INDIVIDUAL_SEGMENT;
	}

	@Reference
	private ContactsEngineClient _contactsEngineClient;

}