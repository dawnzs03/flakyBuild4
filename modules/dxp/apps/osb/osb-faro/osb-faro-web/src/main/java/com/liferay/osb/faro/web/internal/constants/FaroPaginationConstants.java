/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.util.PropsValues;

import java.util.Map;

/**
 * @author Shinn Lok
 */
public class FaroPaginationConstants {

	public static final int CUR_DEFAULT = 1;

	public static final String ORDER_BY_TYPE_ASC = "asc";

	public static final String ORDER_BY_TYPE_DEFAULT = ORDER_BY_TYPE_ASC;

	public static final String ORDER_BY_TYPE_DESC = "desc";

	public static Map<String, Object> getConstants() {
		return _constants;
	}

	private static final Map<String, Object> _constants =
		HashMapBuilder.<String, Object>put(
			"cur", CUR_DEFAULT
		).put(
			"delta", PropsValues.SEARCH_CONTAINER_PAGE_DEFAULT_DELTA
		).put(
			"deltaValues", PropsValues.SEARCH_CONTAINER_PAGE_DELTA_VALUES
		).put(
			"orderAscending", ORDER_BY_TYPE_ASC
		).put(
			"orderDefault", ORDER_BY_TYPE_DEFAULT
		).put(
			"orderDescending", ORDER_BY_TYPE_DESC
		).build();

}