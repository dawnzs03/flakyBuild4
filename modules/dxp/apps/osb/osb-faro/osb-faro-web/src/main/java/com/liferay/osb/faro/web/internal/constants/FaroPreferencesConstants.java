/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

import com.liferay.portal.kernel.util.HashMapBuilder;

import java.util.Map;

/**
 * @author Matthew Kong
 */
public class FaroPreferencesConstants {

	public static final String SCOPE_GROUP = "group";

	public static final String SCOPE_USER = "user";

	public static Map<String, Object> getScopes() {
		return _scopes;
	}

	private static final Map<String, Object> _scopes =
		HashMapBuilder.<String, Object>put(
			SCOPE_GROUP, SCOPE_GROUP
		).put(
			SCOPE_USER, SCOPE_USER
		).build();

}