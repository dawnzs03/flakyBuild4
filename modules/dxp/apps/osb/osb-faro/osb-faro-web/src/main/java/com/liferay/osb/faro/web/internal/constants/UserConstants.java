/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

package com.liferay.osb.faro.web.internal.constants;

import com.liferay.osb.faro.constants.FaroUserConstants;
import com.liferay.portal.kernel.model.RoleConstants;
import com.liferay.portal.kernel.util.HashMapBuilder;

import java.util.Map;

/**
 * @author Matthew Kong
 */
public class UserConstants {

	public static Map<String, String> getRoleNames() {
		return _roleNames;
	}

	public static Map<String, Integer> getStatuses() {
		return _statuses;
	}

	private static final Map<String, String> _roleNames = HashMapBuilder.put(
		"administrator", RoleConstants.SITE_ADMINISTRATOR
	).put(
		"member", RoleConstants.SITE_MEMBER
	).put(
		"owner", RoleConstants.SITE_OWNER
	).build();
	private static final Map<String, Integer> _statuses = HashMapBuilder.put(
		"approved", FaroUserConstants.STATUS_APPROVED
	).put(
		"pending", FaroUserConstants.STATUS_PENDING
	).build();

}