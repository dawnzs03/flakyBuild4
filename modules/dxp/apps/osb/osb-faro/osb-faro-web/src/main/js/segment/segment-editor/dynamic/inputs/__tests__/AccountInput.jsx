import AccountInput from '../AccountInput';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {fromJS} from 'immutable';
import {Property} from 'shared/util/records';
import {PropertyTypes, RelationalOperators} from '../../utils/constants';

jest.unmock('react-dom');

const {EQ} = RelationalOperators;

describe('AccountInput', () => {
	afterEach(cleanup);

	it('should render', () => {
		const {container, getAllByText, getByText} = render(
			<AccountInput
				property={new Property()}
				value={fromJS({criterionGroup: {items: [{operatorName: EQ}]}})}
			/>
		);

		fireEvent.click(getByText('is'));

		expect(getAllByText('is')[1]).toBeTruthy();
		expect(getByText('is not')).toBeTruthy();
		expect(getByText('contains')).toBeTruthy();
		expect(getByText('does not contain')).toBeTruthy();
		expect(getByText('is known')).toBeTruthy();
		expect(getByText('is unknown')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('should render a CustomNumberInput', () => {
		const {queryByText} = render(
			<AccountInput
				property={new Property({type: PropertyTypes.AccountNumber})}
				value={fromJS({criterionGroup: {items: [{operatorName: EQ}]}})}
			/>
		);

		expect(queryByText('is equal to')).not.toBeNull();
	});
});
