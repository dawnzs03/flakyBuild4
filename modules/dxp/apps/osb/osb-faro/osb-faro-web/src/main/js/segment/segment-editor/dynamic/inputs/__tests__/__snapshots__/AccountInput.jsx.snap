// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountInput should render 1`] = `
<div>
  <div
    class="criteria-statement"
  >
    <div
      class="form-group form-group-autofit"
    >
      <div
        aria-hidden="true"
        class="form-group-item entity-name form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      />
      <div
        aria-hidden="true"
        class="form-group-item display-value form-group-item-label form-group-item-shrink"
        data-aria-hidden="true"
      />
      <div
        class="form-group-item form-group-item-shrink"
      >
        <button
          aria-activedescendant="eq"
          aria-expanded="true"
          aria-haspopup="listbox"
          aria-owns="clay-id-2"
          class="form-control form-control-select form-control-select-secondary show"
          role="combobox"
          style=""
          tabindex="0"
          type="button"
        >
          is
        </button>
      </div>
      <div
        aria-hidden="true"
        class="form-group-item"
        data-aria-hidden="true"
      >
        <div
          aria-haspopup="true"
          class="input-group base-select-input-root select-input-root"
        >
          <div
            class="input-group-item"
          >
            <input
              autocomplete="off"
              class="input-root form-control input-group-inset input-group-inset-after"
              placeholder=""
              type="text"
              value=""
            />
            <div
              class="input-group-inset-item input-group-inset-item-after"
            >
              <svg
                class="lexicon-icon lexicon-icon-caret-bottom icon-root"
                role="presentation"
              >
                <use
                  xlink:href="#caret-bottom"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
